common/scripted_effects/wc_vanilla_scripted_effects.txt

```diff
@@ -1259,4 +1259,59 @@ set_up_generated_child_effect = {
 	# Warcraft
 	set_father = event_target:father_target
 	set_mother = event_target:mother_target
+}
+# Handles titles that control religion when converting to secret faith
+handle_control_religion_titles_convert_secret_effect = {
+	# Warcraft
+	if = { # Tidesages are forfeit
+		limit = {
+			religion = tidemother
+			any_demesne_title = {
+				controls_religion = yes
+			}
+		}
+		any_demesne_title = {
+			limit = {
+				controls_religion = yes
+			}
+			destroy_landed_title = THIS
+		}
+	}
+	if = { #The rest are given to a random new character
+		limit = {
+			NOT = { religion = tidemother }
+			any_demesne_title = {
+				controls_religion = yes
+			}
+		}
+		hidden_tooltip = {
+			liege = {
+				save_event_target_as = liege_target
+			}
+			create_character = {
+				age = 38
+				dynasty = random
+				religion = ROOT
+				culture = ROOT
+				random_traits = yes
+			}
+			new_character = {
+				save_event_target_as = new_holder
+			}
+		}
+		any_demesne_title = {
+			limit = {
+				controls_religion = yes
+			}
+			grant_title = event_target:new_holder
+			holder_scope = {
+				any_vassal = {
+					set_defacto_liege = event_target:liege_target
+				}
+			}
+		}
+		hidden_tooltip = {
+			set_defacto_liege = event_target:liege_target
+		}
+	}
```

Comment:
This big chunk of code is never used, since "the rest" (theocracies) can't adopt secret religion